web-app/src/app/router/Router.tsx

```diff
@@ -17,8 +17,34 @@ import TermsAndConditions from '../screens/TermsAndConditions';
 import PrivacyPolicy from '../screens/PrivacyPolicy';
 import Feed from '../screens/Feed';
 import Feeds from '../screens/Feeds';
+import { SIGN_OUT_TARGET } from '../constants/Navigation';
+import {
+  LOGIN_CHANNEL,
+  LOGOUT_CHANNEL,
+  createDispatchChannel,
+} from '../services/channel-service';
+import { useAppDispatch } from '../hooks';
+import { logout } from '../store/profile-reducer';
 
 export const AppRouter: React.FC = () => {
+  const navigateTo = useNavigate();
+  const dispatch = useAppDispatch();
+  const logoutUser = (): void => {
+    dispatch(
+      logout({ redirectScreen: SIGN_OUT_TARGET, navigateTo, propagate: false }),
+    );
+  };
+
+  const loginUser = (): void => {
+    // Refresh the page to ensure the user is authenticated
+    window.location.reload();
+  };
+
+  useEffect(() => {
+    createDispatchChannel(LOGOUT_CHANNEL, logoutUser);
+    createDispatchChannel(LOGIN_CHANNEL, loginUser);
```

Comment:
[question] how come we created a custom subscribe / dispatch system instead of using Redux reducer `dispatch` directly?

Reply:
The redux dispatch doesn't dispatch events across tabs; it's scoped to the current session. We are using redux-persist, but the loading of the state from the local storage is only applicable on redux state rehydration(when the app loads the first time or refreshes). I was looking into other redux libraries that sync states across tabs. However, those libraries make the tabs entirely in sync, making it tricky to have different "states" reflected on each tab. Exanple open different search results or feeds.

Reply:
I added documentation and renamed variable and function names to improve readability.